caffe2/core/predictor.cc

```diff
@@ -48,9 +48,11 @@ void Predictor::run(const TensorVector& inputs, TensorVector* outputs) {
 
   CAFFE_ENFORCE(ws_.RunNet(run_net_.name()));
 
-  outputs->resize(run_net_.external_output_size());
-  for (auto i = 0; i < outputs->size(); ++i) {
-    (*outputs)[i] = extractOutputTensor(&ws_, run_net_.external_output(i));
+  if (outputs != nullptr) {
```

Comment:
Hm, sorry, I was confused. 

What is your usecase, do you run from c++ ? If so, what does this simple predictor give you? You could just run a net the same way. You will have to prepare inputs, but you need to do so in any way.